src/network/core.rs

```diff
@@ -232,6 +232,196 @@ impl Core {
         }
     }
 
+    pub fn macvlan_per_podman_network(
+        per_network_opts: &types::PerNetworkOptions,
+        network: &types::Network,
+        netns: &str,
+    ) -> Result<types::StatusBlock, std::io::Error> {
+        //  StatusBlock response
+        //  StatusBlock response
+        let mut response = types::StatusBlock {
+            interfaces: Some(HashMap::new()),
+        };
+        // Default MACVLAN_MODE to bridge or get from driver options
+        let mut macvlan_mode: u32 = constants::MACVLAN_MODE_BRIDGE;
+        if let Some(options_map) = network.options.as_ref() {
+            if let Some(mode) = options_map.get("mode") {
+                macvlan_mode = core_utils::CoreUtils::get_macvlan_mode_from_string(mode);
+            }
+        }
+
+        // get master interface name
+        let mut master_ifname: String = "".to_string();
+
+        match network.network_interface.as_ref() {
+            None => {
+                if let Ok(ifname) = core_utils::CoreUtils::get_default_route_interface() {
+                    master_ifname = ifname;
+                }
+            }
+            Some(interface) => {
+                master_ifname = interface.to_owned();
+            }
+        }
+
+        // no interface name provided
+        if master_ifname.trim().is_empty() {
```

Comment:
No need fro trim. If there are extra spaces it is a user error, we should not handle this.